scripts/release.js

```diff
@@ -0,0 +1,225 @@
+    name: 'version',
+    message: 'What version do you want to release?',
+    choices: [
+      {
+        major,
+      },
+      {
+        minor,
+      },
+      {
+        patch,
+      },
+    ].map(i => {
+      const key = Object.keys(i)[0];
+      return {
+        key,
+        name: `${key} (${i[key]})`,
+        value: `${i[key]}`,
+      };
+    }),
+  },
+];
+
+/**
+ * Custom errors with detailed explanations
+ */
+const ERRORS = {
+  invalidAndroidEnv: cmd => `Your local environment is not ready to publish native Android bridges.
+
+Check CONTRIBUTING.md to learn how to configure your local environment.
+
+For more details about this error execute:
+  ${colors.yellow(`(cd android && ./gradlew ${cmd})`)}\n`,
+
+  cantPublishAndroid: (
+    cmd,
+    logFileName,
+  ) => `Unable to publish native code for Android bridges.
+To publish it manually execute:
+  ${colors.yellow(`(cd android && ./gradlew ${cmd})`)}
+
+Check CONTRIBUTION.md to learn how to configure your local environment.
+
+For a complete log check:
+  ${colors.yellow(`${logFileName}`)}\n`,
+};
+
+/**
+ * Executes a gradle command inside the android project.
+ * Note that by default this command will suppress stdout and stderr,
+ * you need to provide both in case you want it to be logged.
+ *
+ * @example
+ * ```js
+ * gradleExec(['tasks'], process.stdout, process.stderr)
+ * ```
+ *
+ * @param {Array<String>} cmd the command
+ * @param {WritableStream} stdout stream to redirect standard output
+ * @param {WritableStream} stderr stream to redirect error output
+ *
+ * @returns {Promise} a promise object
+ */
+const gradleExec = (cmd, stdout = null, stderr = null) => {
+  const gradle = spawn(`${androidAppRoot}/gradlew`, cmd, {
+    cwd: androidAppRoot,
+  });
+
+  if (stdout) gradle.stdout.pipe(stdout);
+  if (stderr) gradle.stderr.pipe(stderr);
+
+  return new Promise((resolve, reject) => {
+    gradle.on('close', code => {
+      if (code !== 0) {
+        reject(code);
+      } else {
+        resolve();
+      }
+    });
+  });
+};
+
+/**
+ * Check if the local environment is ready for a release.
+ *
+ * @return {Promise} a promise object
+ */
+function checkEnv() {
+  console.log('🤔  ', '> Checking enviroment');
+  const cmd = `:${androidPackageName}:checkMavenCredentials`;
+  return gradleExec([cmd]).catch(() => {
+    throw new Error(ERRORS.invalidAndroidEnv(cmd));
+  });
+}
+
+/**
+ * Prepare the release package.
+ * This consist manly of copying the contents of `scr` into `dist` and
+ * moving the `js` folder up to the root folder.
+ *
+ * @return {undefined} undefined
+ */
+function prepareRelease() {
+  console.log('🧹   ', '> Cleaning up');
+  execSync(`rm -rf ${distRoot} 2> /dev/null`);
+  execSync(`mkdir ${distRoot}`);
+
+  console.log('📦  ', '> Preparing package');
+  // cp -r src/* is much faster than copying file by file,
+  // to filter out files you don't want to publish edit `.npmignore`.
+  execSync(`cp -r ${srcRoot}/* ${distRoot}`);
```

Comment:
Should this be abstracted into a `npm run build` task?

Reply:
Yeah, good point. Will be better to test it.

Reply:
Done